src/test/java/com/linkedin/kafka/clients/utils/tests/EmbeddedBroker.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils.tests;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FileUtils;
+import org.apache.kafka.common.protocol.SecurityProtocol;
+import org.apache.kafka.common.utils.Time;
+
+
+public class EmbeddedBroker implements AutoCloseable {
+  private final static Class<?> KAFKA_CONFIG_CLASS;
+  private final static Class<?> KAFKA_SERVER_CLASS;
+  private final static Class<?> SCALA_OPTION_CLASS;
+  private final static Class<?> SCALA_SEQ_CLASS;
+  private final static Class<?> SCALA_ARRAY_BUFFER_CLASS;
+  private final static Constructor<?> CONFIG_CTR;
+  private final static Constructor<?> SERVER_CTR;
+  private final static Constructor<?> ARR_BUF_CTR;
+  private final static Method STARTUP_METHOD;
+  private final static Method SHUTDOWN_METHOD;
+  private final static Method AWAIT_SHUTDOWN_METHOD;
+  private final static Object EMPTY_OPTION;
+  private final static Pattern LISTENER_PATTERN = Pattern.compile("\\s*(\\w+)://([\\w\\d]+):(\\d+)\\s*");
```

Comment:
This won't match IPv6 addresses which someone might want to use if they were testing that IPv6 worked.

Reply:
See https://www.ietf.org/rfc/rfc2732.txt

Reply:
Probably the \\s* at the beginning and end is not needed.